dashboard/test/models/user_test.rb

```diff
@@ -5473,6 +5473,70 @@ def hide_scripts_in_sections(section1, section2)
     assert_equal new_us_state, student.reload.us_state
   end
 
+  describe 'strict password requirements by country' do
+    let(:strict_country) {User::PASSWORD_STRICT_COUNTRIES.first}
+
+    before do
+      allow(DCDO).to receive(:get).with('strict-password-country', false).and_return(true)
+      allow(DCDO).to receive(:get).with('migration_service_enabled', false).and_return(false)
+    end
+
+    describe 'when creating a teacher in a strict country' do
+      context 'when using a short password' do
+        let(:user) {build(:teacher, country_code: strict_country, password: 'short', password_confirmation: 'short')}
+        it 'rejects passwords' do
+          _(user).wont_be :valid?
+        end
+
+        it 'adds an error message' do
+          user.valid?
+          _(user.errors[:password]).must_include 'is too short (minimum is 14 characters)'
+        end
+      end
+      context 'when using a short password' do
+        let(:user) {create(:teacher, country_code: strict_country, password: 'longlongpassword', password_confirmation: 'longlongpassword')}
+
+        it 'accepts passwords that are at least 14 characters' do
+          _(user).must_be :valid?
+        end
+
+        it 'has no password errors for long password' do
+          user.valid?
+          _(user.errors[:password]).must_be :empty?
+        end
+      end
+    end
+
+    describe 'when updating a teacher in a strict country' do
+      let(:teacher) {build(:teacher, country_code: strict_country)}
```

Comment:
nit: To simplify a bit the tests, we could include a subject block:
```suggestion
      subject(:update_teacher_password) {teacher.update(password: password, password_confirmation: password)
      let(:teacher) {build(:teacher, country_code: strict_country)}
```

Reply:
I think you would also then have to `let` define the `password` and `password_confirmation` in order for those to be available for the helper... Not to mention, what is `teacher` here if you remove the `let` that defines it. I don't think I've ever really seen `subject` with an argument (TIL) and had to look up examples. `subject` in general is a bit of a shorthand for `let(...)` when it is referring to the main class being tested. There's a universe where you never use a `let` and use the implicit `subject` but I generally dislike such shorthand in favor for a kind of 'consise verbosity' in tests. For instance, this code having the name `teacher` is nice in these tests instead of ALL tests using the term `subject`.

You could (I THINK) keep the `let(:teacher)` and change the `result = teacher.update` to something like `subject!(:updated_teacher) {teacher.update(...) }` inside both `it` blocks and then you'd have just the `_(teacher.errors)` and remove the `_(result).must_equal false`/`true` (which is already implied by the presence/absence of errors)

I think that's a little more confusing than the current tests, though. Curious what the tests for look like using `subject` as a whole to compare, though.